### PR TITLE
Collapsible: typescript

### PR DESCRIPTION
**Pull request by danethurber**
diff is messier than most....sorry

i updated the directory structure to match our other components and the hook that was in `rc` is now used by the main component and exported

### CODE REVIEW

File: packages/verticaltabs/src/react/list.tsx [hunk 1]
```diff
@@ -24,8 +24,9 @@ interface CollapsibleListProps extends ListProps {
   collapsed?: boolean
 }
 export const CollapsibleList: React.FC<CollapsibleListProps> = props => {
+  const { collapsed, ...rest } = props
```
Comment (EdwardIrby) [inline packages/verticaltabs/src/react/list.tsx:27]: nit picky but why not just destructure above?

**Review by EdwardIrby — APPROVED** [review]

**Review by EdwardIrby — CHANGES_REQUESTED** [review]
👇🏿

File: packages/collapsible/tsconfig.build.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    "node_modules",
+    "dist",
+    "**/*.spec.*",
+    "**/*.story.*"
+  ],
+  "compilerOptions": {
+    "declaration": true,
+    "outDir": "dist",
+    "rootDir": "src"
+  }
```
Comment (EdwardIrby) [inline packages/collapsible/tsconfig.build.json:15]: Add strict

File: packages/verticaltabs/src/react/list.tsx (hunk 1 above)
Comment (danethurber) [reply packages/verticaltabs/src/react/list.tsx:27]: personal preference. i don't like it when the line with the component definition breaks onto multiple lines - i think it makes it harder to read. also my editor highlighting can't always parse it. 🤷

**Comment by danethurber** [comment]
the ts build file extends from the other ts file with has strict so it's only needed in the one place

**Pull request merged**